test/pubspec_test.dart

```diff
@@ -592,5 +592,61 @@ executables:
         expect(pubspec.executables['command'], equals('command'));
       });
     });
+
+    group("web", () {
+      test("can be empty", () {
+        var pubspec = new Pubspec.parse('web: {}', sources);
+        expect(pubspec.webCompiler, isEmpty);
+      });
```

Comment:
I don't get a YamlMap in that case, but I added this to the "throws if not a map" test.